### PR TITLE
Merge back security tests

### PR DESCRIPTION
**Pull request by Wadeck**
No related tickets.

Merge back the non-regression tests added for last security release. We are doing that during security release to prevent unexpected merge conflict to ease the process.

ℹ️ Only changes applied to the tests are about:
- assertThat from JUnit moved to the one from Hamcrest due to the deprecation
- a change related to Locale for Digester2Test
- some cleanup of Before methods that has to be integrated

### Proposed changelog entries

* (skip)

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [n/a] (If applicable) Jira issue is well described
- [n/a] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [n/a] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/test/java/hudson/util/Digester2Test.java
Comment (Wadeck) [inline core/src/test/java/hudson/util/Digester2Test.java:47]: Had to update this test as it was failing on my machine working with French default :)

File: test/src/test/java/jenkins/model/JenkinsTest.java
Comment (timja) [inline test/src/test/java/jenkins/model/JenkinsTest.java:729]: this test is failing on CI
https://github.com/jenkinsci/jenkins/pull/5288/checks?check_run_id=1904565850

Comment (Wadeck) [reply test/src/test/java/jenkins/model/JenkinsTest.java:729]: Thanks Tim for the notice. Some tests were failing locally but I hoped it was just due to my local setup (and continuous modifications). Will adjust it soon-ish 👍

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
